sorting/radix_sort2.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ * @brief Algorith of [Radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+ * @author [Suyash Jaiswal](https://github.com/Suyashjaiswal)
+ * @details
+ * Sort the vector of integers using radix sort i.e. sorting digit by digit
+ using [Counting Sort](https://en.wikipedia.org/wiki/Counting_sort) as subroutine.
+ * Running time of radix is O(d*(n+b)) where b is the base for representing
+ numbers and d in the max digits in input inegers and n is number of integers.
+ * consider example for n = 5, aray elements = 432,234,143,332,123
+ *sorting digit by digit
+ *sorting according to
+ *1) 1st digit place
+ *=> 432, 332, 143, 123, 234
+ *
+ *2) 2nd digit place
+ *=> 123, 432, 332, 234, 143
+ *
+ *3) 3rd digit place
+ *=> 123, 143, 234, 332, 432
+ *
+ *using count sort at each step, which is stable.
+ *stable => already sorted according to previous digits.
+*/
+
+// header files
+#include <iostream>  // for io operations
+#include <vector>    // for std::vector
+#include <algorithm> // for collection of functions
+#include <assert.h>  // for a macro called assert which can be used to verify assumptions  
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+    /**
+    * @namespace radix_sort
+    * @brief Functions for [Radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+    * algorithm
+    */
+    namespace radix_sort {
+        /**
+        * @brief Function to sort vector according to current digit using stable
+        * sorting.
+        * @returns std::vector sorted till ith digit
+        */
+        std::vector<int> step_ith(int cur_digit,const std::vector<int>& ar) {  // sorting according to current digit.
```

Comment:
Consider using `uint64_t` or `int64_t` rather than `int` ([reference](https://en.cppreference.com/w/cpp/types/integer)). Check other parts of the code.